### PR TITLE
Extra properties and include other dashboards in the dash.yaml 

### PR DESCRIPTION
**Pull request by keymon**
Added support to specify extra properties  using the key ':graph_properties'. These properties can be and used in
the graphs as normal ruby code. Also allow include all the graphs specified in other dashboard (or any subdirectory in the dashboards directory).

Example::

 :name: Java core metrics.
 :description: Different java process metrics.
 :include: [ "jvm/heap" ]
 :graph_properties:
  :environment: dev
  :servers: [ server1, server2, server3, server4  ]

And in the Graph:

 title       "Heap memory for #{environment} machines"
 timezone    "Europe/London"
 vtitle      "size"
 linewidth   2
 linemode    "connected"
 description "Java Memory Usage in all machines in #{environment}"

 field :heapmax, :alias => "Heap Max",
                 :data  => "sumSeries(servers.java.caspercore.#{environment}.#{servers[0]}.jvm-heap.HeapMemoryUsage_max)"

 servers.each do |servername|
   field "heap_used_#{servername}".to_sym,
   :alias => "Heap Used #{servername}",
   :data  => "servers.java.caspercore.#{environment}.#{servername}.jvm-heap.HeapMemoryUsage_used"
 end

### CODE REVIEW

File: lib/graphite_graph.rb [hunk 1]
```diff
@@ -41,7 +42,7 @@ def defaults
                    :minor_grid_line_color => nil,
                    :area => :none}.merge(@overrides)
   end
-
+
```
Comment (nstielau) [inline lib/graphite_graph.rb:45]: AAAAAAAAAAAHH!!

File: lib/gdash/dashboard.rb [hunk 2]
```diff
@@ -18,14 +18,46 @@ def initialize(short_name, dir, graph_width=500, graph_height=250)
...
 
-      graphs = Dir.entries(directory).select{|f| f.match(/\.graph$/)}
+      graphs = list_graphs(directories)
 
-      graphs.sort.map do |graph|
-        {:name => File.basename(graph, ".graph"), :graphite => GraphiteGraph.new(File.join(directory, graph), {:height => height, :width => width})}
+      graphs.keys.sort.map do |graph_name|
+        {:name => graph_name, 
+         :graphite => GraphiteGraph.new(graphs[graph_name], 
+                      {:height => height, :width => width},
```
Comment (nstielau) [inline lib/gdash/dashboard.rb]: Can we simply use the overrides here instead of passing in a new parameter?

File: lib/gdash/dashboard.rb [hunk 3]
```diff
@@ -18,14 +18,46 @@ def initialize(short_name, dir, graph_width=500, graph_height=250)
...
+      else
+        raise "Invalid value from includes"
+      end
+
+      directories = includes.map { |d|
+        File.join(directory, '..', '..', d)
+      }
+      directories << directory
 
-      graphs = Dir.entries(directory).select{|f| f.match(/\.graph$/)}
```
Comment (nstielau) [inline lib/gdash/dashboard.rb:34]: What about Dir.glob("**/.graph") or something to make this nice and clean?

File: lib/graphite_graph.rb (hunk 1 above)
Comment (keymon) [inline lib/graphite_graph.rb:45]: Is that a horrified scream??? Came on! I've seen worse code! :)

File: lib/gdash/dashboard.rb (hunk 3 above)
Comment (keymon) [inline lib/gdash/dashboard.rb:34]: I would need the location of the graph_templates, and I do not have it here. I think that in that moment it needed a mayor refactor... call me lazy.

File: lib/graphite_graph.rb (hunk 1 above)
Comment (nstielau) [inline lib/graphite_graph.rb:45]: First it's two spaces.  Then global variables.  Then It's a slippery slope ;)

**Comment by nstielau** [comment]
Love the idea, for some of the exact things you mention in the Pull.  I also have a use case to load global config (hacking with .json snippets right now), which might integrate with something like this.  I'll see if I can get some code up soon so we can compare.

**Comment by keymon** [comment]
Cool! This was just a quick hack. I guess everybody solves the problem using chef/puppet templates or similar.

File: lib/gdash/dashboard.rb (hunk 2 above)
Comment (keymon) [inline lib/gdash/dashboard.rb]: "overrides" is overrode somewhere else, I think when you click on graph details. Also it was used in the missing_method and maybe other people/patches expected it. Having a new attribute with defaults was harmless.

**Comment by keymon** [comment]
This patch is no compatible with the original code.

**Pull request closed without merging**